packages/flutter/lib/src/material/text_field.dart

```diff
@@ -298,6 +334,26 @@ class TextField extends StatefulWidget {
   /// {@macro flutter.widgets.editableText.enableInteractiveSelection}
   final bool enableInteractiveSelection;
 
+  /// Called when the user taps on this textfield.
+  ///
+  /// The textfield builds a [GestureDetector] to handle input events like tap,
+  /// to trigger focus requests, to move the caret, adjust the selection, etc.
+  /// Handling some of those events by wrapping the textfield with a competing
+  /// GestureDetector is problematic.
+  ///
+  /// To unconditionally handle taps, without interfering with the textfield's
+  /// internal gesture detector, provide this callback.
+  ///
+  /// If the textfield is created with [enabled] false, taps will not be
+  /// recognized.
+  ///
+  /// To be notified when the textfield gains or loses the focus, provide a
+  /// [focusNode] and add a listener to that.
+  ///
+  /// To listen to arbitrary pointer events without competing with the
+  /// textfield's internal gesture detector, use a [Listener].
+  final GestureTapCallback onTap;
```

Comment:
This will be called twice in a double tap. Is that ok?

Reply:
I haven't looked at the "next PR" yet but if we're timing taps to detect double taps, then we don't need to call this more than once for a series of taps?